File: src/main/java/com/example/today_i_baekjoon/repository/CommitRepository.java
package com.example.today_i_baekjoon.repository;

import com.example.today_i_baekjoon.domain.Commit;
import com.example.today_i_baekjoon.domain.User;
import org.springframework.data.jpa.repository.JpaRepository;

import java.time.LocalDate;
import java.util.List;

public interface CommitRepository extends JpaRepository<Commit, Long> {

    boolean existsByCommitUrl(String commitUrl);

    List<Commit> findAllByCommitDateBetween(LocalDate start, LocalDate end);
    List<Commit> findAllByCommitDateBetweenAndUser(LocalDate start, LocalDate end, User user);
    
    List<Commit> findAllByCommitDate(LocalDate date);
}

File: src/main/kotlin/com/gidskql6671/today_i_baekjoon/repository/CommitRepository.kt
package com.gidskql6671.today_i_baekjoon.repository

import com.gidskql6671.today_i_baekjoon.domain.Commit
import com.gidskql6671.today_i_baekjoon.domain.User
import org.springframework.data.jpa.repository.JpaRepository
import java.time.LocalDate

interface CommitRepository: JpaRepository<Commit, Long> {

    fun existsByCommitUrl(commitUrl: String): Boolean

    fun findAllByCommitDateBetween(start: LocalDate, end: LocalDate): List<Commit>
    fun findAllByCommitDateBetweenAndUser(start: LocalDate, end: LocalDate, user: User): List<Commit>

    fun findAllByCommitDate(date: LocalDate): List<Commit>
}